tasks/tests/unit/test_base.py

```diff
@@ -223,6 +229,98 @@ def test_wrap_up_dbsession_invalid_nothing_works(self, mocker):
+        mock_config_helper(mocker, configs={"setup.timeseries.enabled": False})
+        mock_commit = mocker.patch("tasks.base.django_transaction.commit")
+        task = BaseCodecovTask()
+        task._commit_django()
+        assert mock_commit.call_args_list == [call()]
+
+    def test_rollback_django_with_timeseries(self, mocker):
+        mock_config_helper(mocker, configs={"setup.timeseries.enabled": True})
+        mock_rollback = mocker.patch("tasks.base.django_transaction.rollback")
+        task = BaseCodecovTask()
+        task._rollback_django()
+        assert mock_rollback.call_args_list == [call(), call("timeseries")]
+
+    def test_rollback_django_without_timeseries(self, mocker):
+        mock_config_helper(mocker, configs={"setup.timeseries.enabled": False})
+        mock_rollback = mocker.patch("tasks.base.django_transaction.rollback")
+        task = BaseCodecovTask()
+        task._rollback_django()
+        assert mock_rollback.call_args_list == [call()]
+
+    def test_run_success_commits_both_orms(self, mocker, dbsession):
+        mock_django_commit = mocker.patch("tasks.base.BaseCodecovTask._commit_django")
+        mock_wrap_up = mocker.patch("tasks.base.BaseCodecovTask.wrap_up_dbsession")
+        mock_django_rollback = mocker.patch(
+            "tasks.base.BaseCodecovTask._rollback_django"
+        )
+        mock_dbsession_rollback = mocker.patch.object(dbsession, "rollback")
+        mock_get_db_session = mocker.patch(
+            "tasks.base.get_db_session", return_value=dbsession
+        )
+
+        task = SampleTask()
+        task.run()
+
+        assert mock_django_commit.call_args_list == [call()]
+        assert mock_wrap_up.call_args_list == [call(dbsession)]
+
+        assert mock_django_rollback.call_count == 0
+        assert mock_dbsession_rollback.call_count == 0
+
+    def test_run_db_errors_rollback(self, mocker, dbsession, celery_app):
+        mock_django_commit = mocker.patch("tasks.base.BaseCodecovTask._commit_django")
+        mock_django_rollback = mocker.patch(
+            "tasks.base.BaseCodecovTask._rollback_django"
+        )
+        mock_dbsession_rollback = mocker.patch.object(dbsession, "rollback")
+        mock_wrap_up = mocker.patch("tasks.base.BaseCodecovTask.wrap_up_dbsession")
+        mock_get_db_session = mocker.patch(
+            "tasks.base.get_db_session", return_value=dbsession
+        )
+
+        task = SampleTaskWithArbitraryError(IntegrityError("", {}, None))
+        registered_task = celery_app.register_task(task)
+        task = celery_app.tasks[registered_task.name]
+        task.apply()
+
+        assert mock_django_rollback.call_args_list == [call()]
+        assert mock_dbsession_rollback.call_args_list == [call()]
+
+        assert mock_django_commit.call_args_list == [call()]
+        assert mock_wrap_up.call_args_list == [call(dbsession)]
+
+    def test_run_sqlalchemy_error_rollback(self, mocker, dbsession, celery_app):
```

Comment:
Should this test assert something about sqlalchemy? Or should it say django?

Reply:
`SQLAlchemyError` is the base class. `IntegrityError` is also an example of `SQLAlchemyError` but it has a separate `except` branch so it's a separate test

will add a comment or something explaining